lib/kvs/kvs_cursor.c

```diff
@@ -390,9 +390,9 @@ ikvs_curcache_timer_cb(ulong arg)
  * we have to touch while walking the tree.
  */
 static HSE_ALWAYS_INLINE uint64_t
-ikvs_curcache_key(const uint64_t gen, const char *prefix, const u64 pfxhash, const bool reverse)
+ikvs_curcache_key(const uint64_t gen, const bool reverse)
 {
-    return (gen << 24) | (pfxhash & 0xfffffau) | ((!!prefix) << 1) | reverse;
+    return (gen << 63) | reverse;
```

Comment:
This should dramatically reduce the size of the run-time cursor cache, as well as speed up lookups.